tsc/scripts/git_count.py

```diff
@@ -0,0 +1,495 @@
+        if "NOTFOUND" in comp:
+            print("{} :{}".format(id, comp))
+        
+    
+    
+#
+# Util Function: Use committer's email to find his company name
+#
+def use_email_to_find_company(email, special_return=""):
+
+    #
+    # Step 1: check if email has some keyword for easy identify
+    #
+    ret, comp = company_name_check(email)
+    if ret:
+        return True, comp
+    
+    if "noreply.github.com" in email:
+        s = email.split("@")
+        gid = s[0]
+        # use empty string for email to avoid endless loop
+        comp = get_company_from_id(gid, "")
+        return True, comp
+    
+    if special_return != "":
+        return False, special_return.upper()
+
+    # should not reach here
+    return False, "NONE"
+
+#
+# Fix up company name
+#
+def fixup_companyname(comp):
+    comp = comp.replace("@", "")
+    comp = comp.upper()
+    c = comp.split()
+    comp = c[0]
+    comp = comp.replace(",", "")
+    ret, fix= company_name_check(comp)
+    if ret:
+        comp = fix
+    return comp
+
+#
+# Fix up commiter's name, no space, no (), no "
+#
+def fixup_name(name):
+    name = name.strip()
+    name = name.replace(" ", "+")
+    name = name.replace("(", "")
+    name = name.replace(")", "")
+    name = name.replace("'", "")        
+    return name
+    
+#
+# Pull workspace
+#
+def pull_repros():
+    for repro in repros:
+        cmd = "git clone https://github.com/sonic-net/{}".format(repro)
+        subprocess.check_output(cmd, shell=True)
+
+
+#
+# Use git shortlog to count company commit contributions
+#        
+def count_repro_commits_via_short_log(repro, years):
+
+    # Step 1: use git shortlog  to collect commit count, name and email
+    os.chdir(repro)
+    output = subprocess.check_output("git shortlog --since=\"{} years ago\" -sne".format(years), shell=True)
+    infoarray = output.decode('utf-8').splitlines()
+    count = 0
+    cct = 0
+    for info in infoarray:
+        count = count + 1
+        s = info.split("\t")
+
+        # commit count
+        ccount = int(s[0])
+        cct = cct+ ccount
+        
+        # commiter name
+        nearray = s[1].split("<")
+        name = fixup_name(nearray[0])
+
+        #email
+        email = nearray[1].replace(">","")
+        
+        # Check if there is a cached info for githubid to company mapping entry
+        ret, comp = is_entry_in_id2company(name)
+        if ret:
+        #if ret and "NOTFOUND_" not in comp:
+            # Found cached info, no need to trigger lookup
+            update_company_commit_counts(comp, ccount)
+        else:
+            # Step 2: Check email first for easy cases
+            ret, comp = company_name_check(email)
+            if not ret:
+                # Step 3: use name to find githubid,then use github id to find profile.
+                # Try to get company information from profile
+                url = "https://api.github.com/search/users?q={}".format(name)
```

Comment:
Search by name is wrong. There might be many GitHub users with the same name. The search should be based on a unique key (e.g. email)